VAMobile/src/screens/HealthScreen/Appointments/UpcomingAppointments/UpcomingAppointmentDetails.tsx

```diff
@@ -104,7 +105,14 @@ const UpcomingAppointmentDetails: FC<UpcomingAppointmentDetailsProps> = ({ route
       title: getTranslation(isCovidVaccine ? 'upcomingAppointments.covidVaccine' : AppointmentTypeToID[appointmentType], t),
       startTime: getEpochSecondsOfDate(startDateUtc),
       endTime: getEpochSecondsOfDate(endTime),
-      location: name || '',
+      location:
+        isIOS() && lat && long
```

Comment:
I find nested ternaries hard to read, especially when combined with booleans expressions. Would recommend a refactor (curious if others agree)

Reply:
I agree, I think refactoring this into a helper function and separating the logic would help with readability and maintainability